### ISSUE TITLE
SDDM must not parse messages received from PAM

### ISSUE THREAD

**Issue by darktemplarbasealt** [sddm/sddm#1235]
In function PamBackend::converse sddm receives `int n` and `const struct pam_message **msg` as arguments.
https://github.com/sddm/sddm/blob/c675273bfce0bf9f976f50bb3b9f4d0cc8dc0e7c/src/helper/backend/PamBackend.cpp#L292

`struct pam_message` has following contests:

https://github.com/linux-pam/linux-pam/blob/b429ea18b1c9c8953df5169c6a453b4255a6f23d/libpam/include/security/_pam_types.h#L248-L251

```
struct pam_message {
    int msg_style;
    const char *msg;
};
```

Currently sddm upon receiving such message tries to parse pam_message->msg field and act depending on it's contents. I think that sddm must not parse this message. As far as I can tell, these messages from PAM are intended only for printing or displaying, not for parsing. sddm should process requests received from PAM as they are received only using pam_message->msg_style field, without creating more input fields in advance of PAM requests. This will probably change workflow of SDDM login or password change processes, make them similar to gdm or lightdm: instead of showing all inputs at once, it would show new input for each request from PAM.

Why these changes may be needed despite those UI changes? There are few issues with current approach of parsing PAM messages and taking preemptive actions based on parse results:

- First issue is localization. Parsing fails when PAM sends localized messages.
- If PAM rewords messages, parsing might fail for previously working USE-case. For example, if suddenly word 'password' would be replaced in PAM message with a synonym, sddm would have to be updated or would fail to work.
- If PAM sends unexpected message for USE-case not expected by sddm, sddm fails to work. For example see https://github.com/sddm/sddm/issues/472 from @Mikaela.

In all these cases sddm fails due to parsing message from PAM but would work fine if sddm didn't parse it but just displayed received message and provided an input field of requested type for each received request of type `PAM_PROMPT_ECHO_OFF` or `PAM_PROMPT_ECHO_ON`.

**Comment by darktemplarbasealt** [comment]
What do you think about such changes to sddm? If it sounds good, I can try implementing it.

**Comment by hoehnth** [comment]
Will be addressed in [PR-776](https://github.com/sddm/sddm/pull/776). Had the same problem and removed string comparisons with localization support for PAM messages.

**Comment by darktemplarbasealt** [comment]
It's quite big PR. It was created long time ago and it changes a lot besides current issue. Is it going to be merged at some point of time?

**Comment by hoehnth** [comment]
Yes, big PR because handles PAM localization + (renew) expired passwords. Don't know merge time. It's up to the maintainers decision.
